addons/mail/static/src/web/chatter.js

```diff
@@ -190,6 +190,7 @@ export class Chatter extends Component {
                     );
                 } else {
                     this.state.showAttachmentLoading = false;
+                    this.state.isAttachmentBoxOpened = this.props.isAttachmentBoxVisibleInitially && this.attachments.length > 0;
```

Comment:
I believe the condition should be `||` instead of `&&`?

Reply:
I tested some more with the `||` condition and there are two different behaviours:

- in the contact app, where the view for the chatter has no options set (therefore `isAttachmentBoxVisibleInitially` is `false`) the attachment box will be open initially if there are attachments

- however, in the recruitment app, the view for the chatter says `options="{'open_attachments': True}`, in which case `isAttachmentBoxVisibleInitially` is `true`, and the attachment box will be open, even if there are no attachments

Looking at this test, for example: https://github.com/odoo/odoo/blob/ab678ea7618ca1aee5a6ab31b2de59c87fcce32c/addons/mail/static/tests/web/attachment_box_tests.js#L178
It looks like the attachment box should be initially closed, even when there are attachments. (see the `await click(...)` on line 207)

Reply:
This is a `&&` for me: `props.isAttachmentBoxVisibleInitially` decides what we want, but it's dismissed when there are no attachments (as in that case, it cannot be open as it needs at least 1 attachment to be open)